currencies/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::{marker, result};
+use sr_primitives::traits::StaticLookup;
+use srml_support::{decl_event, decl_module, decl_storage, traits::Get};
+// FIXME: `srml-` prefix should be used for all srml modules, but currently `srml_system`
+// would cause compiling error in `decl_module!` and `construct_runtime!`
+// #3295 https://github.com/paritytech/substrate/issues/3295
+use srml_system::{self as system, ensure_signed};
+
+use traits::{BasicCurrency, BasicCurrencyExtended, MultiCurrency, MultiCurrencyExtended};
+
+mod mock;
+mod tests;
+
+type BalanceOf<T> = <<T as Trait>::MultiCurrency as MultiCurrency<<T as srml_system::Trait>::AccountId>>::Balance;
+type CurrencyIdOf<T> = <<T as Trait>::MultiCurrency as MultiCurrency<<T as srml_system::Trait>::AccountId>>::CurrencyId;
+type ErrorOf<T> = <<T as Trait>::MultiCurrency as MultiCurrency<<T as srml_system::Trait>::AccountId>>::Error;
+
+type AmountOf<T> = <<T as Trait>::MultiCurrency as MultiCurrencyExtended<<T as srml_system::Trait>::AccountId>>::Amount;
+
+pub trait Trait: system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+	type MultiCurrency: MultiCurrencyExtended<Self::AccountId>;
+	type GetNativeCurrencyId: Get<CurrencyIdOf<Self>>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Currencies {
+
+	}
+}
+
+decl_event!(
+	pub enum Event<T> where
+		<T as system::Trait>::AccountId,
+		Balance = BalanceOf<T>,
+		CurrencyId = CurrencyIdOf<T>
+	{
+		/// Currency transfer success (currency_id, from, to, amount)
+		Transferred(CurrencyId, AccountId, AccountId, Balance),
+	}
+);
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		fn deposit_event() = default;
+
+		/// Transfer some balance to another account.
+		pub fn transfer(
```

Comment:
should have transfer handle both native and non native currency and use `T:: NativeCurrency ` for native transfer